transit_odp/timetables/utils.py

```diff
@@ -234,3 +235,277 @@ def get_line_description_based_on_direction(row: pd.Series) -> str:
+        "common_name",
+        "stop_sequence",
+        "vehicle_journey_code",
+        "departure_time",
+        "atco_code",
+    ]
+    df_vehicle_journey_operating = df_vehicle_journey_operating[columns_to_keep]
+    df_vehicle_journey_operating = df_vehicle_journey_operating.drop_duplicates()
+
+    # Get the unique sorted vehicle journey codes based on the departure time
+    df_vehicle_journey_grouped = df_vehicle_journey_operating.groupby(
+        by="vehicle_journey_code"
+    )
+    df_vehicle_journey_grouped = df_vehicle_journey_grouped.agg(
+        {"departure_time": "min"}
+    )
+    df_vehicle_journey_sorted = df_vehicle_journey_grouped.sort_values(
+        by="departure_time"
+    )
+    df_vehicle_journey_sorted = df_vehicle_journey_sorted.reset_index()
+    vehicle_journey_codes_sorted = df_vehicle_journey_sorted["vehicle_journey_code"]
+
+    # Add a new column stating the vehicle journey order based on vehicle journey codes order
+    vehicle_journey_codes_sorted_dict = {
+        code: index for index, code in enumerate(vehicle_journey_codes_sorted)
+    }
+    df_vehicle_journey_operating[
+        "vehicle_journey_order"
+    ] = df_vehicle_journey_operating["vehicle_journey_code"].replace(
+        vehicle_journey_codes_sorted_dict
+    )
+
+    # Sort the data frame based on the vehicle journey and the stop sequence
+    df_vehicle_journey_operating = df_vehicle_journey_operating.sort_values(
+        by=["vehicle_journey_order", "stop_sequence"]
+    )
+    df_vehicle_journey_operating.to_csv("df_vehicle_journey_operating_sorted.csv")
+    # TODO: Add the logic for creating the final dataframe of timetable
+
+    return pd.DataFrame()
+
+
+def filter_df_serviced_org_operating(
+    target_date: str,
+    df_serviced_org_working_days: pd.DataFrame,
+    all_exception_vehicle_journey: np.ndarray,
+) -> Tuple[List, List]:
+    """
+    Get the vehicle journeys based on the serviced organisation
+    working days and the operating/non-operating execptions
+
+    :return: DataFrame
+    Return the filtered dataframe based on the serviced organisation
+    """
+
+    df_serviced_org_working_days.to_csv("serviced.csv")
+
+    # Remove the vehicle journey which are not running on target date (nonoperating exception)
+    df_serviced_org_working_days = df_serviced_org_working_days[
+        ~df_serviced_org_working_days["vehicle_journey_id"].isin(
+            all_exception_vehicle_journey
+        )
+    ]
+
+    # Get the operating and non-operating working days
+    df_service_operating = df_serviced_org_working_days[
+        df_serviced_org_working_days["operating_on_working_days"]
+    ]
+    df_service_nonoperating = df_serviced_org_working_days[
+        ~df_serviced_org_working_days["operating_on_working_days"]
+    ]
+
+    # Find the service which are operating within range of start and end date
+    df_service_operating = df_service_operating[
+        (target_date > df_serviced_org_working_days.start_date)
```

Comment:
Target date should be greater than or equal to the start date and the same with the end date where equals sign needs to be added

Reply:
Done